adapter/ipListChecker/aspect.go

```diff
@@ -27,62 +27,21 @@ import (
 
 	"github.com/golang/glog"
 	"gopkg.in/yaml.v2"
-	"istio.io/mixer/pkg/adapter"
 )
 
-// AspectConfig is used to configure an adapter.
-type AspectConfig struct {
-	adapter.AspectConfig
-
-	// ProviderURL is the URL where to find the list to check against
-	ProviderURL string
-
-	// RefreshInterval determines how often the provider is polled for
-	// an updated list.
-	RefreshInterval time.Duration
-
-	// TimeToLive indicates how long to keep a list before discarding it.
-	// Typically, the TTL value should be set to noticeably longer (> 2x) than the
-	// refresh interval to ensure continued operation in the face of transient
-	// server outages.
-	TimeToLive time.Duration
-}
-
 type aspectState struct {
 	backend         *url.URL
 	atomicList      atomic.Value
 	fetchedSha      [sha1.Size]byte
 	refreshInterval time.Duration
 	ttl             time.Duration
 	closing         chan bool
+	fetchError      error
+	client          http.Client
 }
 
-// newAspect returns a new aspect.
-func newAspect(config *AspectConfig) (adapter.ListChecker, error) {
-	var u *url.URL
-	var err error
-	if u, err = url.Parse(config.ProviderURL); err != nil {
-		// bogus URL format
-		return nil, err
-	}
-
-	a := aspectState{
-		backend:         u,
-		closing:         make(chan bool),
-		refreshInterval: config.RefreshInterval,
-		ttl:             config.TimeToLive,
-	}
-
-	// install an empty list
-	a.setList([]*net.IPNet{})
-
-	// load up the list synchronously so we're ready to accept traffic immediately
-	a.refreshList()
-
-	// crank up the async list refresher
-	go a.listRefresher()
-
-	return &a, nil
+func (a *aspectState) Name() string {
```

Comment:
In the aspectCache, we have fully constructed aspect structs. these structs do not hold a reference to the Adapter  (Builder) that they came from. ImplName is that link, which I think is useful in logging.